pilot/pkg/serviceregistry/kube/controller/util.go

```diff
@@ -89,30 +89,6 @@ func findPortFromMetadata(svcPort v1.ServicePort, podPorts []model.PodPort) (int
 	return 0, fmt.Errorf("no matching port found for %+v", svcPort)
 }
 
-type serviceTargetPort struct {
-	// the mapped port number, or 0 if unspecified
-	num int
-	// the mapped port name
-	name string
-	// a bool indicating if the mapped port name was explicitly set on the TargetPort field, or inferred from k8s' port.Name
-	explicitName bool
-}
-
-func findServiceTargetPort(servicePort *model.Port, k8sService *v1.Service) serviceTargetPort {
-	for _, p := range k8sService.Spec.Ports {
-		// TODO(@hzxuzhonghu): check protocol as well as port
-		if p.Name == servicePort.Name || p.Port == int32(servicePort.Port) {
-			if p.TargetPort.Type == intstr.Int && p.TargetPort.IntVal > 0 {
-				return serviceTargetPort{num: int(p.TargetPort.IntVal), name: p.Name, explicitName: false}
-			}
-			return serviceTargetPort{num: 0, name: p.TargetPort.StrVal, explicitName: true}
-		}
-	}
-	// should never happen
-	log.Debugf("did not find matching target port for %v on service %s", servicePort, k8sService.Name)
-	return serviceTargetPort{num: 0, name: "", explicitName: false}
-}
```

Comment:
IIRC some of this is to handle cases where the targetPort is a name _or_ a number?

And that check is lost with the changes?


Are we sure that's desired?

Reply:
`TargetPort` struct contains both a name and a number

For k8s Service:
- targetPort: 8080 -> TargetPort{Name: "", Number: 8080}
- targetPort: "http" -> TargetPort{Name: "http", Number: 0}

For ServiceEntry:
- ```
  ports:
  - number: 80
    name: http
  ```

-> TargetPort{Name: "http", Number: 0}
- ```
  ports:
  - number: 80
    name: http
    targetPort: 8080
  ```
-> TargetPort{Name: "http", Number: 8080}

There should be no functionality loss as long as we follow this semantics